api/store/create.js

```diff
@@ -0,0 +1,93 @@
+module.exports = createStore
+
+var _ = require('lodash')
+var request = require('request').defaults({json: true})
+
+var addRolePrivilege = require('../utils/add-role-privilege')
+var couchDbDesignDoc = require('../utils/couchdb-design-doc')
+var errors = require('../utils/errors')
+var getRoleOption = require('../utils/get-role-option')
+var hoodieAccessToCouchDbSecurity = require('../utils/hoodie-access-to-couchdb-security')
+var storeExists = require('./exists')
+var toDbId = require('../utils/to-db-id')
+
+function createStore (state, name, options) {
+  return storeExists(state, name)
+
+  .then(function (doesExist) {
+    if (doesExist) {
+      throw errors.CONFLICT
+    }
+
+    var doc = {
+      _id: toDbId(name),
+      access: {
+        read: {
+          role: []
+        },
+        write: {
+          role: []
+        }
+      }
+    }
+
+    if (options) {
+      _.concat(options.access).forEach(function (privilege) {
+        addRolePrivilege(doc.access, getRoleOption(options), privilege)
+      })
+    }
+
+    return state.metaDb.put(doc)
+
+    .then(function () {
+      var db = new state.PouchDB(name)
```

Comment:
I think what  I’ll do is to set `state.encodeName` to a function that will either return the name of the db or encode it, based on whether the passed in `PouchDB` constructor will use the `http` adapter by default. Does that sound good? 

This probably won’t be easy to fix for the PouchDB folks, and if they do it might be a breaking change as people might already encode the name and you’d end up with double-encoding